File: 0189-rotate-array/0189-rotate-array.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {void} Do not return anything, modify nums in-place instead.
 */
var rotate = function(arr, k) {
    const tempArray = Array.from(arr)
    let n = arr.length ;
    for(let i = 0;i<n;i++) {
        arr[(i+k)%n] = tempArray[i];
    }
    return arr ;
};

File: 0189-rotate-array/0189-rotate-array.py
class Solution:
    def rotate(self, arr: List[int], k: int) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        temp = arr.copy()
        n = len(arr)
        for i in range(n) :
            newIndex = (i+k)%n
            arr[newIndex] = temp[i] 
        return arr